src/game/WorldHandlers/ChatHandler.cpp

```diff
@@ -93,7 +100,10 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket& recv_data)
         SendNotification(LANG_UNKNOWN_LANGUAGE);
         return;
     }
-    if (langDesc->skill_id != 0 && !_player->HasSkill(langDesc->skill_id))
+
+    //prevent cheating, by sending LANG_UNIVERSAL
+    if ((langDesc->lang_id == LANG_UNIVERSAL && !sWorld.getConfig(CONFIG_BOOL_ALLOW_TWO_SIDE_INTERACTION_CHAT) && GetSecurity() == SEC_PLAYER) ||
+         langDesc->skill_id != 0 && !_player->HasSkill(langDesc->skill_id))
```

Comment:
Here is the code as it is at the moment `(A || B && C)` did you mean `((A || B) && C)` or did you mean `(A || (B && C))`?

Here is an example of a compiler warning generated by such code: https://travis-ci.org/Rochet2/smallfolk_cpp/jobs/202645497#L314-L317